### PR TITLE
Pv devel

### PR DESCRIPTION
**Pull request by andr2000**

### CODE REVIEW

File: sound/drivers/xen-sndfront.c [hunk 1]
```diff
@@ -0,0 +1,2127 @@
...
+	EVTCHNL_STATE_CONNECTED,
+	EVTCHNL_STATE_SUSPENDED,
+};
+
+struct xdrv_evtchnl_info {
+	struct xdrv_info *drv_info;
+	struct xen_sndif_front_ring ring;
+	int ring_ref;
+	unsigned int port;
+	unsigned int irq;
```
Comment (otyshchenko1) [inline sound/drivers/xen-sndfront.c]: Why unsigned int? Type int is used in both cases "int bind_evtchn_to_irqhandler()" and "notify_remote_via_irq(int)"

File: sound/drivers/xen-sndfront.c [hunk 2]
```diff
@@ -0,0 +1,2127 @@
...
+	if (channel->irq)
+		unbind_from_irqhandler(channel->irq, channel);
+	channel->irq = 0;
+	if (channel->port)
+		xenbus_free_evtchn(drv_info->xb_dev, channel->port);
+	channel->port = 0;
+	/* End access and free the pages */
+	if (channel->ring_ref != GRANT_INVALID_REF)
+		gnttab_end_foreign_access(channel->ring_ref, 0,
+			(unsigned long)channel->ring.sring);
```
Comment (otyshchenko1) [inline sound/drivers/xen-sndfront.c]: channel->ring_ref = GRANT_INVALID_REF;

File: sound/drivers/xen-sndfront.c [hunk 3]
```diff
@@ -0,0 +1,2127 @@
...
+
+static void sdrv_alsa_timer_update(struct sdev_alsa_timer_info *dpcm)
+{
+	unsigned long delta;
+
+	delta = jiffies - dpcm->base_time;
+	if (!delta)
+		return;
+	dpcm->base_time += delta;
+	delta *= dpcm->rate;
```
Comment (otyshchenko1) [inline sound/drivers/xen-sndfront.c]: Are the delta has enough type not to overflow?

File: sound/drivers/xen-sndfront.c (hunk 1 above)
Comment (andr2000) [inline sound/drivers/xen-sndfront.c]: changed to int

File: sound/drivers/xen-sndfront.c [hunk 4]
```diff
@@ -0,0 +1,2127 @@
...
+	channel->state = EVTCHNL_STATE_DISCONNECTED;
+	/* release all who still waits for response if any */
+	channel->resp_status = -XENSND_RSP_ERROR;
+	complete_all(&channel->completion);
+	if (channel->irq)
+		unbind_from_irqhandler(channel->irq, channel);
+	channel->irq = 0;
+	if (channel->port)
+		xenbus_free_evtchn(drv_info->xb_dev, channel->port);
+	channel->port = 0;
```
Comment (andr2000) [inline sound/drivers/xen-sndfront.c]: done, 0 is the correct value, IMO

File: sound/drivers/xen-sndfront.c (hunk 2 above)
Comment (andr2000) [inline sound/drivers/xen-sndfront.c]: done

Comment (andr2000) [inline sound/drivers/xen-sndfront.c]: added some unlikely on somewhat critical path's

**Pull request closed without merging**